source/_posts/react-localstorage.md

```diff
@@ -0,0 +1,382 @@
+* Dark mode feature.
+* User's form input (those that don't contain critical information).
+* Caching the retrieved data from some API (is not recommended if the architecture is not defined very well, there are best approaches).
+* Persisting data in general.
+
+
+## Vanilla JavaScript Style
+
+As we mentioned earlier, `localStorage` is an API that allows you to access a `Storage` object of the document. It stores key-value pairs when the unique type of value admitted is in UTF-16 string format (later, we will see it with a clear example).
+
+`localStorage` allows you to set, erase and retrieve data. In case you need it, this is the [official documentation](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
+
+### Set data to localStorage
+
+To set some data (persist data) in the `localStorage`, we have two ways:
+
+```js
+// Set key-value (both are equivalent, the second could be use to set
+// keys programmatically), allows you to save data
+localStorage.counter = 100;
+// or
+localStorage.setItem('counter', 100);
+```
+
+Try to set a number. Even if it's an object, it will be converted to a string and will be treated as such. To see that case, we have an example:
+
+```js
+// ❌ Set key-value (Wrong way, json object)
+localStorage.user = {
+    name: 'Jonas',
+    age: 12,
+    description: 'a lot of confusing references'
+}
+
+// The result will be "[object Object]" because set method infers the 
+// .toString() value, so try it
+console.log(localStorage.user);
+```
+
+To set objects properly see the next example:
+
+```js
+// If you need to save a JSON object (encode before and decode after)
+localStorage.setItem('user', JSON.stringify({
+    name: 'Jonas',
+    age: 12,
+    description: 'a lot of confusing references'
+}))
+const user = JSON.parse(localStorage.getItem('user'))
+// the output will be a JSON string
+console.log(user);
+```
+
+### Get data from localStorage
+
+To get data we have two ways too.
+
+```js
+// Get key-value (both are equivalent, the second could be use to get
+// keys programmatically), allows you to get the saved data
+let counter = localStorage.counter;
+// or
+counter = localStorage.getItem('counter');
+console.log(counter);
+```
+remember getItem only returns a string value.
+
+### Removing data stored in localStorage
+
+If `localStorage` doesn't expire the unique way to remove its data is using two methods: `removeItem` and `clear`, see it with an example:
+
+```js
+// Remove key-value, allows you to remove data
+localStorage.removeItem('counter');
+// Alternativelly 'delete' works but it happens thanks 
+// to the language and not of the localStorage implementation
+// so be careful, we don't recommend its use
+delete localStorage.counter;
+
+// Cleaning all data from localStorage 
+localStorage.clear();
+```
+
+`removeItem` and clear are different. `removeItem` removes a key-value pair, and `clear` removes all the data inside `localStorage`.
```

Comment:
`clear` ?

Reply:
or even better `removeItem()` and `clear()` ?

Reply:
done :D